### PR TITLE
chore(wdio): port css-variables test suite to webdriverio

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## What is the current behavior?

GitHub Issue Number: N/A

## What is the new behavior?

port the `test/karma/test-app/css-variables` test suite to webdriverio.
    
the suite itself didn't exist prior to this port, so a new test has been
written to attempt to capture the spirit of running this test manually
(spinning up the stencil www app and interacting with the button that's
rendered).
    
git history has been preserved as much as possible here, using `git mv`
to move files from the `karma` dir to `wdio/` in all possible cases.
however, git may fail to mark some files as 'moved' if we change a file
enough that it exceeds git's "sameness threshold".

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing
To verify this worked, I made the tests "red" (fail) for every assertion, then one by one made them pass/green

## Other information
STENCIL-1191

### CODE REVIEW

File: test/wdio/css-variables/cmp.test.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import './variables.css';
```
Comment (rwaskiewicz) [inline test/wdio/css-variables/cmp.test.tsx:1]: These tests didn't exist prior to this port, so I tried to write tests that captured the spirit of the original. Leaving some notes/comments/questions inline

File: test/wdio/css-variables/cmp.test.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+}`;
+
+describe('css-variables', () => {
+  beforeEach(() => {
+    render({
+      template: () => (
+        <>
+          <style>{css}</style>
+
+          <css-variables-no-encapsulation></css-variables-no-encapsulation>
```
Comment (rwaskiewicz) [inline test/wdio/css-variables/cmp.test.tsx:23]: The original test had a second instance of `css-variables-no-encapsulation`. I don't see a reason why we need this, as we don't change anything in `css-variables-no-encapsulation` in the original test (so there's no state change encapsulation to test). LMK if something jumps out at you that I'm missing here

File: test/wdio/css-variables/cmp.test.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+  describe('css-variables-no-encapsulation', () => {
+    it('uses class-local css variables to set the background and color', async () => {
+      const blackLocalElm = await $('.black-local');
+
+      await expect(blackLocalElm).toHaveText('No encapsulation: Black background');
+      await expect(blackLocalElm).toHaveStyle({
+        background: 'none 0% 0% auto repeat padding-box border-box scroll rgba(0, 0, 0, 1)',
+        color: 'rgba(255,255,255,1)',
+      });
```
Comment (rwaskiewicz) [inline test/wdio/css-variables/cmp.test.tsx:40]: I tried to use something like:
```ts
await expect(blackLocalElm.getCssProperty('background')).toBe(...)
```
but the left hand side still returned `none 0% 0% auto repeat padding-box border-box scroll rgba(0, 0, 0, 1)`, so I just left this as-is. LMK if you know if there's a better way to go about this!

File: test/wdio/css-variables/cmp.test.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    it('uses global css variables to set the background, color, and font weight', async () => {
+      const blackGlobalElm = await $('.black-global');
+
+      await expect(blackGlobalElm).toHaveText('No encapsulation: Black background (global style)');
+      await expect(blackGlobalElm).toHaveStyle({
+        background: 'none 0% 0% auto repeat padding-box border-box scroll rgba(0, 0, 0, 1)',
+        color: 'rgba(255,255,255,1)',
+      });
+
+      const fontWeight = await blackGlobalElm.getCSSProperty('font-weight');
```
Comment (rwaskiewicz) [inline test/wdio/css-variables/cmp.test.tsx:52]: AFAICT, this property isn't exposed by `toHaveStyle`, hence the separate assertion

File: test/wdio/css-variables/cmp.test.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        background: 'none 0% 0% auto repeat padding-box border-box scroll rgba(255, 255, 0, 1)',
+        color: 'rgba(0,0,0,1)',
+      });
+    });
+  });
+
+  describe('css-variables-shadow-dom', () => {
+    it("doesn't interfere with global variables", async () => {
+      const shadowRoot = document.querySelector('css-variables-shadow-dom').shadowRoot;
+      const globalShadow = await $('>>>.black-global-shadow');
```
Comment (rwaskiewicz) [inline test/wdio/css-variables/cmp.test.tsx:70]: The `shadowRoot` here is solely for clicking the button.

I tried to do:
```tsx
await $('>>>button').waitForExist()
const button = await $('>>>button');
await button.click();
```

But that didn't seem to work for some reason. LMK if I'm doing something wrong when it comes to piercing the shadow DOM, as it doesn't appear the click was actually happening.

**Review by christian-bromann — COMMENTED** [review]
We can get rid of the sync clicks and pauses with this patch:

```patch
diff --git a/test/wdio/css-variables/cmp.test.tsx b/test/wdio/css-variables/cmp.test.tsx
index b927898ed..8d8bdf4b3 100644
--- a/test/wdio/css-variables/cmp.test.tsx
+++ b/test/wdio/css-variables/cmp.test.tsx
@@ -67,7 +67,6 @@ describe('css-variables', () => {
 
   describe('css-variables-shadow-dom', () => {
     it("doesn't interfere with global variables", async () => {
-      const shadowRoot = document.querySelector('css-variables-shadow-dom').shadowRoot;
       const globalShadow = await $('>>>.black-global-shadow');
 
       await expect(globalShadow).toHaveText('Shadow: Black background (global)');
@@ -78,10 +77,7 @@ describe('css-variables', () => {
       let fontWeight = await globalShadow.getCSSProperty('font-weight');
       expect(fontWeight.value).toEqual(800);
 
-      const button = shadowRoot.querySelector('button');
-      button.click();
-      // pause since this click isn't async
-      await browser.pause(100);
+      await $('css-variables-shadow-dom').$('>>> button').click()
 
       await expect(globalShadow).toHaveText('Shadow: Black background (global)');
       await expect(globalShadow).toHaveStyle({
@@ -93,19 +89,13 @@ describe('css-variables', () => {
     });
 
     it('repaints as a result of changing css variables', async () => {
-      const shadowRoot = document.querySelector('css-variables-shadow-dom').shadowRoot;
-      const innerDiv = await $('>>>.inner-div');
-
+      const innerDiv = $('>>>.inner-div');
       await expect(innerDiv).toHaveStyle({
         background: `none 0% 0% auto repeat padding-box border-box scroll ${browser.isChromium ? 'rgba(255, 0, 0, 1)' : 'rgb(255, 0, 0)'}`,
         color: browser.isChromium ? 'rgba(0,0,255,1)' : 'rgb(0,0,255)',
       });
 
-      const button = shadowRoot.querySelector('button');
-      button.click();
-      // pause since this click isn't async
-      await browser.pause(100);
-
+      await $('css-variables-shadow-dom').$('>>> button').click();
       await expect(innerDiv).toHaveStyle({
         background: `none 0% 0% auto repeat padding-box border-box scroll ${browser.isChromium ? 'rgba(0, 128, 0, 1)' : 'rgb(0, 128, 0)'}`,
         color: browser.isChromium ? 'rgba(0,0,255,1)' : 'rgb(0,0,255)',
@@ -113,50 +103,32 @@ describe('css-variables', () => {
     });
 
     it('changes the text as a result of changing css variables', async () => {
-      const shadowRoot = document.querySelector('css-variables-shadow-dom').shadowRoot;
-      const innerDiv = await $('>>>.inner-div');
-
+      const innerDiv = $('>>>.inner-div');
       await expect(innerDiv).toHaveText('Shadow: Red background');
-
-      const button = shadowRoot.querySelector('button');
-      button.click();
-      // pause since this click isn't async
-      await browser.pause(100);
-
+      await $('css-variables-shadow-dom').$('>>> button').click();
       await expect(innerDiv).toHaveText('Shadow: Green background');
     });
 
     it("doesn't change the font weight as a result of changing css variables", async () => {
-      const shadowRoot = document.querySelector('css-variables-shadow-dom').shadowRoot;
-      const innerDiv = await $('>>>.inner-div');
-
+      const innerDiv = $('>>>.inner-div');
       let fontWeight = await innerDiv.getCSSProperty('font-weight');
       expect(fontWeight.value).toEqual(400);
 
-      const button = shadowRoot.querySelector('button');
-      button.click();
-      // pause since this click isn't async
-      await browser.pause(100);
-
+      await $('css-variables-shadow-dom').$('>>> button').click();
       fontWeight = await innerDiv.getCSSProperty('font-weight');
       await expect(fontWeight.value).toEqual(400);
     });
 
     it("doesn't alter a second instance of the element", async () => {
-      // grab the first shadow root so we can click it's button
-      const shadowRoot = document.querySelectorAll('css-variables-shadow-dom')[0].shadowRoot;
       // grab a reference to the second inner-div class to verify it's background does not change
       const innerDiv = await $$('css-variables-shadow-dom')[1].$('>>>.inner-div');
-
       await expect(innerDiv).toHaveStyle({
         background: `none 0% 0% auto repeat padding-box border-box scroll ${browser.isChromium ? 'rgba(255, 0, 0, 1)' : 'rgb(255, 0, 0)'}`,
         color: browser.isChromium ? 'rgba(0,0,255,1)' : 'rgb(0,0,255)',
       });
 
-      const button = shadowRoot.querySelector('button');
-      button.click();
-      // pause since this click isn't async
-      await browser.pause(100);
+      const button = $('css-variables-shadow-dom').$('>>> button');
+      await button.click();
 
       await expect(innerDiv).toHaveStyle({
         background: `none 0% 0% auto repeat padding-box border-box scroll ${browser.isChromium ? 'rgba(255, 0, 0, 1)' : 'rgb(255, 0, 0)'}`,
```

File: test/wdio/css-variables/cmp.test.tsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
+import './variables.css';
```
[the remark this answers is not in the document]
Comment (rwaskiewicz) [reply test/wdio/css-variables/cmp.test.tsx:1]: Yeah, it applies the styling that we assert against here - https://github.com/ionic-team/stencil/pull/5468/files#diff-fb56270e504fe95a8257a7606ea0a656a44d488ef070fc1382e576d61dfe3291R62

**Comment by rwaskiewicz** [comment]
Thanks @christian-bromann much cleaner :chefs-kiss:

File: test/wdio/css-variables/cmp.test.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (christian-bromann) [reply test/wdio/css-variables/cmp.test.tsx:1]: ohh .. right, Vite automatically attaches it as a link tag:

```html
<style type="text/css" data-vite-dev-id="/path/to/stencil/test/wdio/css-variables/variables.css">
:root {
  --link-background: yellow;
}
</style>
```

**Review by christian-bromann — APPROVED** [review]
👍

**Review by tanner-reits — APPROVED** [review]

**Pull request merged**